packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -179,8 +179,9 @@ declare const Astro: {
 						throw e;
 					}
 					let hydrationTimeStart;
+					const hydrator = this.hydrator(this);
 					if (process.env.NODE_ENV === 'development') hydrationTimeStart = performance.now();
-					await this.hydrator(this)(this.Component, props, slots, {
+					await hydrator(this.Component, props, slots, {
```

Comment:
I'm not sure if this *actually* does something, but when testing locally, I was getting times that made a bit more sense and were generally more stable 🤷‍♀️